server/application/terminal.go

```diff
@@ -54,6 +57,13 @@ func (s *terminalHandler) getApplicationClusterRawConfig(ctx context.Context, a
 	return clst.RawRestConfig(), nil
 }
 
+// isValidKubernetesResourceName checks for valid kubernetes resource names
+func isValidKubernetesResourceName(name string) bool {
+	// quick check to ensure we aren't passing along anything unsafe
+	isQualified := validation.IsQualifiedName(name)
```

Comment:
I think the correct validator for container names and namespaces would be "DNS label" (https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.19/#container-v1-core): https://github.com/kubernetes/apimachinery/blob/28c7554c5cff9dcf360f7fe734df1554b97c5a67/pkg/util/validation/validation.go#L188

And the validation for pod names would be "DNS subdomain": https://github.com/kubernetes/kubernetes/blob/976a940f4a4e84fe814583848f97b9aafcdb083f/pkg/apis/core/validation/validation.go#L241